### PR TITLE
Fix DDR field inconsistency in mixed refs builds

### PR DESCRIPTION
**Pull request by sharon-wang**
Fixes: https://github.com/eclipse/openj9/issues/11469
OMR changes: https://github.com/eclipse/omr/pull/5758

- Remove `!defined(OMR_OVERRIDE_COMPRESS_OBJECT_REFERENCES)` from preprocessor guard of `_compressObjectReferences` field
- Make `_compressObjectReferences` a `protected` field in header files that declare it -- keeping it as a private field while removing the OVERRIDE check causes clang to error on osx (`-Wunused-private-field`)

I ran the failing test `cmdLineTester_callsitedbgddrext_openj9_0` with these changes and omr changes on the failing platforms  `ppc64_aix_mixed` and `s390x_linux_mixed`, and I've built locally on mac -- all are passing.

Related: #8878, #11166

### CODE REVIEW

File: runtime/gc_base/ObjectAccessBarrier.hpp
Comment (gacholio) [inline runtime/gc_base/ObjectAccessBarrier.hpp:62]: This is public not protected.

File: runtime/gc_base/ObjectAccessBarrier.cpp
Comment (keithc-ca) [inline runtime/gc_base/ObjectAccessBarrier.cpp:70]: Missing `(`.

File: runtime/gc_include/ObjectAccessBarrierAPI.hpp
Comment (keithc-ca) [inline runtime/gc_include/ObjectAccessBarrierAPI.hpp:57]: Please keep `const` here.

File: runtime/gc_include/ObjectAccessBarrierAPI.hpp
Comment (keithc-ca) [inline runtime/gc_include/ObjectAccessBarrierAPI.hpp:77]: This won't compile when `!(defined(OMR_GC_COMPRESSED_POINTERS) && defined(OMR_GC_FULL_POINTERS))`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/gc_include/ObjectAccessBarrierAPI.hpp
Comment (sharon-wang) [reply runtime/gc_include/ObjectAccessBarrierAPI.hpp:77]: I'm going to move the `protected` section so that the field doesn't come first. Reordering should be okay to do per https://github.com/eclipse/omr/pull/5758#discussion_r560411683

**Comment by sharon-wang** [comment]
Addressed comments and tested against the failing tests -- they are now passing.

**Review by keithc-ca — APPROVED** [review]

**Review by gacholio — APPROVED** [review]

**Comment by sharon-wang** [comment]
#### Jenkins shortnames for mixed refs CMake builds:

`aixmxd`: ppc64_aix_mixed
`zlinuxmxd`: s390x_linux_mixed
`plinuxmxd`: ppc64le_linux_mixed
`xlinuxmxd`: x86-64_linux_mixed
`winmxd`: x86-64_windows_mixed
`osxmxd`: x86-64_mac_mixed
`alinux64mxd`: aarch64_linux_mixed

---

These changes need to be run with https://github.com/eclipse/omr/pull/5758, or https://github.com/eclipse/omr/pull/5758 needs to get merged and become available on the openj9 branch first.

**Comment by gacholio** [comment]
jenkins test sanity zlinux jdk8

**Comment by gacholio** [comment]
jenkins test sanity plinuxmxd jdk11

**Comment by gacholio** [comment]
Ah didn't realize the changes were dependent. I'll relaunch testing after the OMR code is merged.

**Comment by gacholio** [comment]
Actually, there's no reason they would be dependent, and the PR builds here are working fine, so I'll merge this when the testing completes.

**Comment by gacholio** [comment]
jenkins compile win jdk8 depends https://github.com/eclipse/omr/pull/5758

**Comment by pshipton** [comment]
Note that plinux always passed, it was zlinux & AIX (Big Endian) platforms that failed.

**Pull request merged**
